lang/cs/Org.Apache.REEF.Network/Group/Task/Impl/GroupCommClient.cs

```diff
@@ -71,18 +73,21 @@ public sealed class GroupCommClient : IGroupCommClient
             }
 
             networkService.Register(new StringIdentifier(taskId));
+        }
 
+        public void WaitingForRegistration(CancellationTokenSource cancellationSource = null)
+        {
             try
             {
                 foreach (var group in _commGroups.Values)
                 {
-                    group.WaitingForRegistration();
+                    group.WaitingForRegistration(cancellationSource);
                 }
             }
-            catch (ReefRuntimeException e)
+            catch (SystemException e)
```

Comment:
Yes, I meant that both of these exceptions can be un-inherited from `ReefRuntimeException`